server/database_access/retrieve-buildings.js

```diff
@@ -0,0 +1,85 @@
+import { Buildings } from '/imports/database/collections';
+import { getDistanceFromLatLonInKm } from '../utilities/distance-utils';
+import { mongoToLocalTime, stringToLocalTime } from '../utilities/date-time-utils';
+
+/**
+ * Gets the numBuildings closest buildings to a given lat/lon at a given time on a given day,
+ * where none of the returned buildings have zero unoccupied rooms.
+ * @param numBuildings
+ * @param lat
+ * @param lon
+ * @param time
+ * @param day
+ * @returns {Array.<T>}
+ */
+export function getClosestAvailableBuildings(numBuildings, lat, lon, time, day) {
+    const availableBuildingsArray = [];
+    Buildings.find({}).forEach(function (building) {
+        let availableRooms = [];
+        building.Rooms().forEach(function(room) {
+            let clashingRoomSlot = false;
```

Comment:
Rename clash to something that implies the room is occupied, not clashing with another roomslot.